File: NOVEMBER-2025/10-11-2025/p3542.py
#python

class Solution(object):
    def minOperations(self, nums):
        s = [0]
        res = 0
        for a in nums:
            while s and s[-1] > a:
                s.pop()
            if not s or s[-1] < a:
                res += 1
                s.append(a)
        return res


#python3


class Solution:
    def minOperations(self, nums):
        stack = [0] * (len(nums) + 1)
        top = 0
        ans = 0

        for num in nums:
            while stack[top] > num:
                top -= 1
                ans += 1
            if stack[top] != num:
                top += 1
                stack[top] = num

        return ans + top

File: NOVEMBER-2025/10-11-2025/p3542.rs
const N: usize = 100_000;

impl Solution {
    pub fn min_operations(nums: Vec<i32>) -> i32 {
        let n = nums.len();
		let mut stk = [0; N];
		let mut m = 0;
		let mut ans = 0;

		for x in nums {
			while m > 0 && stk[m-1] > x {
				ans += 1;
				m -= 1;
			}
			if x != 0 && (m <= 0 || stk[m-1] < x) {
				stk[m] = x;
				m += 1;
			}
		}

		ans += m;

		ans as i32
    }
}